.github/workflows/release-dataset.yml

```diff
@@ -0,0 +1,41 @@
+name: Release dataset to production DB
+
+on:
+  workflow_dispatch:
+    branches: 
+      - main
+    inputs: 
+      megaset-name:
+        description: "Enter the name of the megaset to release every dataset within"
+        required: false
+      dataset-id: 
+        description: "Enter the ID in the format [NAME]_v[VERSION] to release a single dataset"
+        required: false
```

Comment:
if you default these both to "" (ie empty string) I don't think you need a if statement at all